libs/Microsoft.MixedReality.WebRTC/ISignalingService.cs

```diff
@@ -0,0 +1,221 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+
+namespace Microsoft.MixedReality.WebRTC
+{
+    public enum SignalingType
+    {
+        SdpOffer,
+        SdpAnswer,
+        IceCandidate
+    }
+
+    public class SignalingMessage
+    {
+        public string SessionId;
+        public SignalingType Type;
+        public string Payload;
+    }
+
+    interface ISignalingService : IDisposable
+    {
+        /// <summary>
+        /// Start the service and make the local process visible as `localEndPointId`.
+        /// </summary>
+        void Start(string localEndPointId);
+
+        /// <summary>
+        /// Stop the service.
+        /// </summary>
+        void Stop();
+
+        /// <summary>
+        /// Get the endpoints that are currently active.
+        /// </summary>
+        IEnumerable<string> RemoteEndPoints { get; }
```

Comment:
This is a bit confusing. WebRTC signaling is about a peer-to-peer connection, there are never multiple peers. Is this a discovery/matchmaking concept rather? Is this interface really about signaling (the process of connecting 2 known peers), or about discovery/matchmaking (the process of selecting the 2 peers before signaling)?

Reply:
Good point - this is about both. Maybe we want to split it? It doesn't seem trivial as finding endpoints and communicating with them are related even if not the same thing. Ideas?